cpp/src/arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/odbc_impl/encoding_utils.h

```diff
@@ -77,4 +77,37 @@ inline size_t ConvertToSqlWChar(const std::string& str, SQLWCHAR* buffer,
   }
 }
 
+/// @brief Convert buffer of SqlWchar to standard string
+/// @param wchar_msg[in] SqlWchar to convert
+/// @param msg_len[in] Number of characters in wchar_msg
+/// @return wchar_msg in std::string format
+inline std::string SqlWcharToString(SQLWCHAR* wchar_msg, SQLSMALLINT msg_len = SQL_NTS) {
+  if (wchar_msg == nullptr) {
+    return std::string();
+  }
+
+  // Size of SQLWCHAR depends on the platform
+  size_t wchar_size = sizeof(SQLWCHAR);
+
+  // Assert that size of SQLWCHAR and wchar_t are the same
+  assert(wchar_size == sizeof(wchar_t));
```

Comment:
Might be able to use static_assert() here instead of assert() to get a compile-time failure instead of a runtime failure.